modApiServer/src/org/aion/api/server/rpc3/Web3EntryPoint.java

```diff
@@ -66,17 +72,24 @@ public String call(String requestString){
             err = e.getError();//Don't log this error since it may already be logged elsewhere
         }
         catch (RPCException e){
-            logger.debug("Request failed due to an RPC exception: ", e);
+            logger.debug("Request failed due to an RPC exception: {}", e.getMessage());
             err = e.getError();
         }
         catch (Exception e){
-            logger.debug("Call to {} failed.", request==null? "null":request.method);
-            logger.debug("Request failed due to an internal error: ", e);
-            err= InternalErrorRPCException.INSTANCE.getError();
+            logger.error("Call to {} failed.", request==null? "null":request.method);
+            logger.error("Request failed due to an internal error: ", e);
+            err= new InternalErrorRPCException(
+                e.getClass().getSimpleName()+":"+e.getMessage()).getError(); // Prefer this over
+                                                                // creating an instance of RpcError
         }
         final String resultString = ResponseConverter
             .encodeStr(new Response(id, resultUnion, err, VersionType.Version2));
-        logger.debug("Produced response: {}", resultString);
+
+        if (logger.isDebugEnabled() && stopwatch != null) {
```

Comment:
If the logger got pass from the AionLogger, then doesn't need this check anymore.

Reply:
I suppose here the check could be `stopwatch != null`